### PR TITLE
more micro optimizations

### PR DESCRIPTION
**Pull request by dfa1**
Hello @bbakerman @andimarek,

I did some more profiling with YourKit and found these new "opportunities":

image

JMH benchmark:

master
```
Benchmark                                    Mode  Cnt   Score   Error  Units
BenchMark.benchMarkSimpleQueriesThroughput  thrpt   15  47.091 ± 1.469  ops/s
BenchMark.benchMarkSimpleQueriesAvgTime      avgt   15  21.572 ± 1.113  ms/op
```

with FetchedValue fix
```
Benchmark                                    Mode  Cnt   Score   Error  Units
BenchMark.benchMarkSimpleQueriesThroughput  thrpt   15  47.069 ± 0.646  ops/s
BenchMark.benchMarkSimpleQueriesAvgTime      avgt   15  20.776 ± 0.804  ms/op
```

with CacheKey fix
```
Benchmark                                    Mode  Cnt   Score   Error  Units
BenchMark.benchMarkSimpleQueriesThroughput  thrpt   15  52.207 ± 1.457  ops/s
BenchMark.benchMarkSimpleQueriesAvgTime      avgt   15  19.707 ± 0.459  ms/op
```

IMHO is a small but important improvement for all users of the library!

### CODE REVIEW

**Comment by dfa1** [comment]
@bbakerman `PropertyFetcherBenchMark` is also improving on my laptop.

Master branch:
```
# JMH version: 1.25
# VM version: JDK 1.8.0_77, Java HotSpot(TM) 64-Bit Server VM, 25.77-b03
# VM invoker: /Library/Java/JavaVirtualMachines/jdk1.8.0_77.jdk/Contents/Home/jre/bin/java
# VM options: -javaagent:/Users/dfa/Library/Application Support/JetBrains/Toolbox/apps/IDEA-U/ch-0/203.5981.155/IntelliJ IDEA.app/Contents/lib/idea_rt.jar=62641:/Users/dfa/Library/Application Support/JetBrains/Toolbox/apps/IDEA-U/ch-0/203.5981.155/IntelliJ IDEA.app/Contents/bin -Dfile.encoding=UTF-8
# Warmup: 2 iterations, 5 s each, 3 calls per op
# Measurement: 3 iterations, 10 s each, 4 calls per op
# Timeout: 10 min per iteration
# Threads: 1 thread, will synchronize iterations
# Benchmark mode: Throughput, ops/time
# Benchmark: benchmark.PropertyFetcherBenchMark.benchMarkThroughputDirectClassHierarchy

# Run progress: 0.00% complete, ETA 00:06:40
# Fork: 1 of 5
objc[12979]: Class JavaLaunchHelper is implemented in both /Library/Java/JavaVirtualMachines/jdk1.8.0_77.jdk/Contents/Home/jre/bin/java (0x105abb4c0) and /Library/Java/JavaVirtualMachines/jdk1.8.0_77.jdk/Contents/Home/jre/lib/libinstrument.dylib (0x105b354e0). One of the two will be used. Which one is undefined.
# Warmup Iteration   1: 1208527.970 ops/s
# Warmup Iteration   2: 1436059.592 ops/s
Iteration   1: 1155912.403 ops/s
Iteration   2: 1158149.131 ops/s
Iteration   3: 1103212.762 ops/s

```

This branch:
```
# JMH version: 1.25
# VM version: JDK 1.8.0_77, Java HotSpot(TM) 64-Bit Server VM, 25.77-b03
# VM invoker: /Library/Java/JavaVirtualMachines/jdk1.8.0_77.jdk/Contents/Home/jre/bin/java
# VM options: -javaagent:/Users/dfa/Library/Application Support/JetBrains/Toolbox/apps/IDEA-U/ch-0/203.5981.155/IntelliJ IDEA.app/Contents/lib/idea_rt.jar=62647:/Users/dfa/Library/Application Support/JetBrains/Toolbox/apps/IDEA-U/ch-0/203.5981.155/IntelliJ IDEA.app/Contents/bin -Dfile.encoding=UTF-8
# Warmup: 2 iterations, 5 s each, 3 calls per op
# Measurement: 3 iterations, 10 s each, 4 calls per op
# Timeout: 10 min per iteration
# Threads: 1 thread, will synchronize iterations
# Benchmark mode: Throughput, ops/time
# Benchmark: benchmark.PropertyFetcherBenchMark.benchMarkThroughputDirectClassHierarchy

# Run progress: 0.00% complete, ETA 00:06:40
# Fork: 1 of 5
objc[13043]: Class JavaLaunchHelper is implemented in both /Library/Java/JavaVirtualMachines/jdk1.8.0_77.jdk/Contents/Home/jre/bin/java (0x10db2d4c0) and /Library/Java/JavaVirtualMachines/jdk1.8.0_77.jdk/Contents/Home/jre/lib/libinstrument.dylib (0x10dba74e0). One of the two will be used. Which one is undefined.
# Warmup Iteration   1: 4103756.324 ops/s
# Warmup Iteration   2: 6146158.084 ops/s
Iteration   1: 4577539.556 ops/s
Iteration   2: 4711435.476 ops/s
Iteration   3: 4654780.485 ops/s
```

File: src/main/java/graphql/execution/FetchedValue.java [hunk 1]
```diff
@@ -13,12 +12,12 @@
     private final Object fetchedValue;
     private final Object rawFetchedValue;
     private final Object localContext;
-    private final ImmutableList<GraphQLError> errors;
+    private final List<GraphQLError> errors;
```
Comment (andimarek) [inline src/main/java/graphql/execution/FetchedValue.java:15]: why did you change this do `List`?

**Review by andimarek — COMMENTED** [review]
Looks good but one comment

**Comment by andimarek** [comment]
thanks!

File: src/main/java/graphql/execution/FetchedValue.java (hunk 1 above)
Comment (andimarek) [reply src/main/java/graphql/execution/FetchedValue.java]: changed it myself
